Adds how-to guide on adding authorities to access tokens

ddubson opened the pull request:
Closes issue #542

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+[[how-to-extension-grant-type]]
+= How-to: Add authorities to custom claims in JWT-based access tokens
```
Please update to -> "How-to: Add authorities as custom claims in JWT access tokens"

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+[[how-to-extension-grant-type]]
```
Update to `how-to-custom-claims-authorities`

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+:docs-dir: ..
+:examples-dir: {docs-dir}/examples
+
+This guide demonstrates how to append resource owner authorities to a JSON Web Token (JWT)-based access token.
```
append -> add
JSON Web Token (JWT)-based access token -> JWT access token

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+:examples-dir: {docs-dir}/examples
+
+This guide demonstrates how to append resource owner authorities to a JSON Web Token (JWT)-based access token.
+The term 'authorities' may be described in varying forms such as roles, permissions, or groups of the resource owner.
```
Use double quotes for 'authorities'

may be described in varying forms -> may represent varying forms

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+This guide demonstrates how to append resource owner authorities to a JSON Web Token (JWT)-based access token.
+The term 'authorities' may be described in varying forms such as roles, permissions, or groups of the resource owner.
+
+To make resource owners' authorities available to the resource server, we add custom claims to an access token issued by Spring Authorization Server.
```
resource owners' -> resource owner's

jgrandja commented on docs/src/docs/asciidoc/examples/src/main/java/sample/customClaims/authorities/CustomClaimsWithAuthoritiesConfiguration.java:
```diff
@@ -0,0 +1,43 @@
+		return (context) -> {
+			if (OAuth2TokenType.ACCESS_TOKEN.equals(context.getTokenType())) { // <4>
+				context.getClaims().claims((claims) -> { // <5>
+					var roles = AuthorityUtils.authorityListToSet(context.getPrincipal().getAuthorities())
```
Replace `var` with `Set<String>`

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+
+<1> Define a sample user `user1` within an internal user details service.
+<2> Define a few roles for  `user1`.
+<3> Define `OAuth2TokenCustomizer<JwtEncodingContext>` bean that allows for customizing JWT token claims.
```
bean -> `@Bean`

jgrandja commented on docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:
```diff
@@ -0,0 +1,49 @@
+<6> Extract user roles from the `Principal` object. The role information for internal users is stored as a string prefixed with `ROLE_`, so we strip the prefix here.
+<7> Set custom claim `roles` to the set of roles collected from the previous step.
+
+As a result of this customization, authorities information about the user will be included as a custom claim within an
```
within an access token -> within the access token

jgrandja reviewed (changes requested):
Thanks for the PR @ddubson ! Please see review comments.

Also, please rebase on `main`. Thanks!

ddubson commented:
@jgrandja Thanks for the review! I've addressed the points outlined and rebased onto the latest Antora-based docs structure. Antora rendering looks great!

jgrandja commented:
Thanks for the updates @ddubson ! This is now merged along with a polish commit.
